exchanges/stream/websocket_connection.go

```diff
@@ -67,52 +73,60 @@ func (w *WebsocketConnection) SendJSONMessage(ctx context.Context, data interfac
 }
 
 // SendRawMessage sends a message over the connection without JSON encoding it
-func (w *WebsocketConnection) SendRawMessage(ctx context.Context, messageType int, message []byte) error {
-	return w.writeToConn(ctx, func() error {
+func (w *WebsocketConnection) SendRawMessage(ctx context.Context, epl request.EndpointLimit, messageType int, message []byte) error {
+	return w.writeToConn(ctx, epl, func() error {
 		if w.Verbose {
 			log.Debugf(log.WebsocketMgr, "%v websocket connection: sending message [%s]\n", w.ExchangeName, message)
 		}
 		return w.Connection.WriteMessage(messageType, message)
 	})
 }
 
-func (w *WebsocketConnection) writeToConn(ctx context.Context, writeConn func() error) error {
+func (w *WebsocketConnection) writeToConn(ctx context.Context, epl request.EndpointLimit, writeConn func() error) error {
 	if !w.IsConnected() {
```

Comment:
With the other removal, we could put this after the locks. The other one I'd argue is more important

Reply:
Yeah I did think about this, the only reason I had for keeping the upper one is so routines don't hang consecutively on the rate limit sleep when there is no available connection. When you are writing to the connection it does the check anyway and determines if its connected 🤷 and it's hard to get coverage for it trying to time execution. I won't resolve it just yet, maybe there is something else to consider by other reviews.